scripts/batch-score.coffee

```diff
@@ -0,0 +1,50 @@
+# Description:
+#   Script for batch wise score.
+#
+# Commands:
+#   hubot score bxx
+# 
+# Author:
+#   aseem09
+
+cron = require('node-cron')
+module.exports = (robot) ->
+
+  listOfUsersWithCount = (year) ->
+  
+    sorted = []
+    for own key, user of robot.brain.data.users
+
+      temp2=user[4]
+      user_year=temp2.split('')
+      temp3 = parseInt( user_year[0], 10 );
+
+      if `temp3 == year` > 0
+        sorted.push [user.name, user.msgcount]
+
+    if sorted.length
+      sorted.sort (a, b) ->
+        b[1] - a[1]
+    return sorted
+  
+  # response for score status of any <batch>
+  robot.respond /score b([\w\-_]+)/i, (msg) ->
```

Comment:
`b[\w\-_]` matches any line starting with `b`, containing only alphanumeric chars thereafter. So, I think `b\d\d` could be a better regex since it'll match `b<number><number>` (eg b21, b22).

Reply:
Done.